### PR TITLE
Support unknown in predicted sex

### PR DESCRIPTION
**Pull request by bpblanken**

### CODE REVIEW

File: v03_pipeline/var/test/sex_check/test_imputed_sex.tsv [hunk 1]
```diff
@@ -1,4 +1,4 @@
...
...8b8d/895966ef-c705-4c18-952d-03863243a184/PIE_OGI1433_002628_1.cram.md5sum	gs://datarepo-c41dc160-bucket/907593be-8862-4945-9e70-f758b6448b8d/96ca6d5f-fb23-4102-bb5e-c7bbfd194e1c/PIE_OGI1433_002628_1.cram	ffb50687-165e-425a-a545-c3797d3a28d4	07.021.604.3.7.8	55729ba9-3ce4-47b3-9c3b-1148737ae40f	2024-04-17T15:07:57	99.670000000	gs://datarepo-c41dc160-bucket/907593be-8862-4945-9e70-f758b6448b8d/30f8e208-5d2d-4ce8-b835-695b5ed673f4/PIE_OGI1433_002628_1.mapping_metrics.csv	DNA:DNA Genomic	41.910000000	Whole Blood:Whole Blood	PT-25BR5	Pass	PDO-32756	92.920000000	97.990000000	Unknown	P-WG-0139	2017-05-19 04:00:00	Unknown	RP-3062	gs://datarepo-c41dc160-bucket/907593be-8862-4945-9e70-f758b6448b8d/1641d1b2-1035-4cc3-9c8b-0c8cb430f56b/PIE_OGI1433_002628_1.vcf.gz.tbi	gs://datarepo-c41dc160-bucket/907593be-8862-4945-9e70-f758b6448b8d/f5ba2708-899e-42e8-b287-fdf72c2e404d/PIE_OGI1433_002628_1.vcf.gz.md5sum	gs://datarepo-c41dc160-bucket/907593be-8862-4945-9e70-f758b6448b8d/e925ee5d-a75e-471f-adfd-2756c8690069/PIE_OGI1433_002628_1.vcf.gz	156149580126.000000000	gs://datarepo-c41dc160-bucket/907593be-8862-4945-9e70-f758b6448b8d/df076bc5-9db8-44f0-a3fe-f693370634cc/PIE_OGI1433_002628_1.vc_metrics.csv
```
Comment (bpblanken) [inline v03_pipeline/var/test/sex_check/test_imputed_sex.tsv:4]: Changed `Male` -> `Unknown`

File: v03_pipeline/var/test/sex_check/test_imputed_sex_unexpected_value.tsv [hunk 2]
```diff
@@ -1,2 +1,2 @@
...
...atarepo-556a9c15-bucket/2a4202b0-93f5-4ebe-8d2b-fd4cfb2b881d/ec41ec06-673f-4fe2-a063-23dc5fe1dcce/GLE-5774-3-3.cram.md5sum	gs://datarepo-556a9c15-bucket/2a4202b0-93f5-4ebe-8d2b-fd4cfb2b881d/aad0e270-2ad5-4f39-b968-9b4beafeb5cc/GLE-5774-3-3.cram	a4b04a39-9234-4028-a155-442c4acf12a0	07.021.604.3.7.8	ce74d94c-c33d-49d7-85c9-5f3cbd08aff7	2024-04-17T15:02:46	99.800000000	gs://datarepo-556a9c15-bucket/2a4202b0-93f5-4ebe-8d2b-fd4cfb2b881d/c3a9e6f2-4c68-410b-823d-46ca406e5061/GLE-5774-3-3.mapping_metrics.csv	DNA:DNA Genomic	35.300000000	Whole Blood:Whole Blood	PT-24OHM	Pass	PDO-32755	96.320000000	97.340000000	UNKNOWN	P-WG-0139	2017-04-12 04:00:00	UNKNOWN	RP-3061	gs://datarepo-556a9c15-bucket/2a4202b0-93f5-4ebe-8d2b-fd4cfb2b881d/c71cd2a1-c789-4715-9ebc-dbfc40d9f2e2/GLE-5774-3-3.vcf.gz.tbi	gs://datarepo-556a9c15-bucket/2a4202b0-93f5-4ebe-8d2b-fd4cfb2b881d/957a99cb-c9a9-4fc5-a0ec-53f9e461469e/GLE-5774-3-3.vcf.gz.md5sum	gs://datarepo-556a9c15-bucket/2a4202b0-93f5-4ebe-8d2b-fd4cfb2b881d/df520949-5f2b-4976-9d46-80d1cc299813/GLE-5774-3-3.vcf.gz	133253714921.000000000	gs://datarepo-556a9c15-bucket/2a4202b0-93f5-4ebe-8d2b-fd4cfb2b881d/2e98e51b-9394-4e64-977f-e9010a4e16dc/GLE-5774-3-3.vc_metrics.csv
```
Comment (bpblanken) [inline v03_pipeline/var/test/sex_check/test_imputed_sex_unexpected_value.tsv:2]: Changed `Unknown` -> `UNKNOWN`

File: v03_pipeline/lib/misc/family_loading_failures_test.py [hunk 3]
```diff
@@ -72,12 +72,12 @@ def test_build_sex_check_lookup(self):
         self.assertEqual(
             build_sex_check_lookup(ht, hl.dict({'ROS_006_18Y03226_D1': 'remapped_id'})),
             {
-                'remapped_id': Sex.MALE,
-                'ROS_006_18Y03227_D1': Sex.MALE,
-                'ROS_006_18Y03228_D1': Sex.MALE,
-                'ROS_007_19Y05919_D1': Sex.MALE,
-                'ROS_007_19Y05939_D1': Sex.FEMALE,
-                'ROS_007_19Y05987_D1': Sex.MALE,
+                'remapped_id': Sex.FEMALE,
```
Comment (bpblanken) [inline v03_pipeline/lib/misc/family_loading_failures_test.py:75]: I changed these to better align with the pedigree... but no functional difference.

File: v03_pipeline/lib/misc/family_loading_failures.py [hunk 4]
```diff
@@ -175,10 +175,16 @@ def get_families_failed_sex_check(
...
-                Sex.UNKNOWN,
-            }:  # NB: Unknown samples in pedigree are excluded from sex check.
+            # NB: Both Unknown samples in pedigree and Unknown
+            # samples in the predicted_sex are precluded from
+            # failing the sex check.
+            if (
+                sex_check_lookup[sample_id] == Sex.UNKNOWN  # noqa: PLR1714
+                or family.samples[sample_id].sex == Sex.UNKNOWN
+            ):
+                continue
```
Comment (matren395) [inline v03_pipeline/lib/misc/family_loading_failures.py:185]: STOPGAP for the implementation I'd wanted - could we logger.info() the samples excluded from sex checks for this ?

File: v03_pipeline/lib/misc/io_test.py [hunk 5]
```diff
@@ -45,15 +45,15 @@ def test_import_imputed_sex(self) -> None:
...
+                hl.Struct(s='abc_3', predicted_sex='U'),
             ],
         )
 
     def test_import_imputed_sex_unexpected_value(self) -> None:
         ht = import_imputed_sex(TEST_IMPUTED_SEX_UNEXPECTED_VALUE)
         self.assertRaisesRegex(
             hl.utils.java.HailUserError,
-            'Found unexpected value Unknown in imputed sex file',
+            'Found unexpected value UNKNOWN in imputed sex file',
```
Comment (matren395) [inline v03_pipeline/lib/misc/io_test.py:56]: I'm turned around on this - I thought UNKNOWN is expected now ?

Comment (bpblanken) [reply v03_pipeline/lib/misc/io_test.py:56]: This one is testing an "unexpected" value.  `Unknown` is now expected, but `UNKNOWN` is not.  `Unknown` was literally the value we chose to test as being unexpected previously just by happenstance 🤣

File: v03_pipeline/lib/misc/family_loading_failures.py (hunk 4 above)
Comment (bpblanken) [reply v03_pipeline/lib/misc/family_loading_failures.py:185]: 👍

File: v03_pipeline/lib/misc/io_test.py (hunk 5 above)
Comment (bpblanken) [reply v03_pipeline/lib/misc/io_test.py:56]: Just to make it clear that `Unknown` is expected and `UNKNOWN` will still throw an error.  We're going to continue to be non-lenient here.

Comment (matren395) [reply v03_pipeline/lib/misc/io_test.py:56]: Hmm I was about to suggest something a bit clearer, but this probably is good for being explicit about leniency

File: v03_pipeline/lib/misc/family_loading_failures.py [hunk 6]
```diff
@@ -175,10 +178,18 @@ def get_families_failed_sex_check(
...
-            }:  # NB: Unknown samples in pedigree are excluded from sex check.
+            # NB: Both Unknown samples in pedigree and Unknown
+            # samples in the predicted_sex are precluded from
+            # failing the sex check.
+            if (
+                sex_check_lookup[sample_id] == Sex.UNKNOWN  # noqa: PLR1714
+                or family.samples[sample_id].sex == Sex.UNKNOWN
+            ):
+                logger.info(
+                    f'Encountered sample with Unknown sex excluded from sex check: {sample_id}'
```
Comment (matren395) [inline v03_pipeline/lib/misc/family_loading_failures.py:189]: and these are gonna be strings, right, not a Hail string type thtat's goint to print as <Hail.String type at 07x40194578> or whatever , yea ? If so then lgtm

**Review by matren395 — APPROVED** [review]

File: v03_pipeline/lib/misc/family_loading_failures.py (hunk 6 above)
Comment (bpblanken) [reply v03_pipeline/lib/misc/family_loading_failures.py:189]: Ya, everything is `hl.eval`ed by this point.  The [signature of the](https://github.com/broadinstitute/seqr-loading-pipelines/pull/900/files/bb440635cbdfcad1e0114b9b70803e81a0553069#diff-81bda30e9edc96dbaf6b9533d4db17f8afa17df35de4b37c3c05e1856e6779dbL125) `build_sex_check_lookup` is correct I think!

**Pull request merged**